lib/Ohmbrewer_RIMS.cpp

```diff
@@ -28,57 +28,57 @@ Ohmbrewer::Pump* Ohmbrewer::RIMS::getRecirculator() const {
 /**
  * Constructor
  * @param id The Sprout ID to use for this piece of Equipment
- * @param pins The list of physical pins this Equipment is attached to
+ * @param tubePins[ temp busPin ; heating powerPin ; heating controlPin ]
+ * @param tunBus - mash tun temperature pin
+ * @param pumpPins[powerPin ; controlPin ]
  */
-Ohmbrewer::RIMS::RIMS(int id, std::list<int>* pins) : Ohmbrewer::Equipment(id, pins) {
-    // TODO: Figure out how to properly set the components of the RIMS the constructors
-    std::list<int>* fakePins = new std::list<int>(1,-1);
-    _tube = new Thermostat(1, fakePins);
-    _tunSensor = new TemperatureSensor(1, fakePins);
-    _recirc = new Pump(1,fakePins);
+Ohmbrewer::RIMS::RIMS(int id, int tubePins[], int tunBus, int pumpPins[] ) : Ohmbrewer::Equipment(id) {
```

Comment:
You should have a check to ensure that the arrays that are passed in are appropriately sized. If it fails, maybe initialize with the ID-only constructor and/or send an error out using the Publisher class?